python/graphstorm/model/lm_embed.py

```diff
@@ -880,3 +882,158 @@ def forward(self, input_feats, input_nodes):
                 input_feats[ntype] = lm_feat
 
         return super(GSLMNodeEncoderInputLayer, self).forward(input_feats, input_nodes)
+
+class GSLMNodeEncoderInputLayer4GraphFromMetaData(GSNodeEncoderInputLayer):
+    """ The node encoder input layer with language model (LM) supported for all nodes
+    in a heterogeneous graph.
+
+    This input layer treats node features in the same way as the ``GSNodeEncoderInputLayer``.
+    In addition, the input layer reloads LM layer and projection layer on nodes with textual 
+    features and generate embeddings.
+
+    Parameters
+    ----------
+    g: DistGraph
+        The input GSDglDistGraphFromMetadata.
+    node_lm_configs: LM config
+        A list of language model configurations.
+    feat_size : dict of int or dict of list of ints
+        The original feat sizes of each node type in the format of {ntype: size}.
+        If a node have multiple feature groups, it is in the format of {ntype: [size, size, ...]}.
+    embed_size : int
+        The output embedding size.
+    activation : callable
+        The activation function. Default: None.
+    dropout : float
+        The dropout parameter. Default: 0.0.
+    use_node_embeddings : bool
+        Whether to use the node embeddings for individual nodes even when node features are
+        available. Default: False.
+    force_no_embeddings : list of str
+        The list node types that are forced to not use learnable embeddings. Default: None
+    """
+    def __init__(self,
+                 g,
+                 node_lm_configs,
+                 feat_size,
+                 embed_size,
+                 activation=None,
+                 dropout=0.0,
+                 use_node_embeddings=False,
+                 force_no_embeddings=None):
+        assert node_lm_configs is not None and len(node_lm_configs) > 0, \
+            "language model configurations must be provided"
+
+        self.node_lm_configs = node_lm_configs
```

Comment:
I didn't find where this `self.node_lm_configs` is used in other places. If not, can remove this line.